File: Java/T5/codigos2122/ProyectoDialogos/app/src/main/java/com/example/proyectodialogos/dialog/DialogoConfirmacion.java
package com.example.proyectodialogos.dialog;

import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

public class DialogoConfirmacion extends DialogFragment {

    private Context context;

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        this.context = context;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

        AlertDialog.Builder alertDialog = new AlertDialog.Builder(getActivity());
        alertDialog.setTitle("Diálogo de confirmacion");
        alertDialog.setMessage("Diálogo de confirmacion para ver como funciona. ¿Estas seguro?");
        alertDialog.setPositiveButton("OK", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                Toast.makeText(context, "Selección positiva", Toast.LENGTH_SHORT).show();
            }
        });
        alertDialog.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                Toast.makeText(context, "Selección negativa", Toast.LENGTH_SHORT).show();

            }
        });
        alertDialog.setNeutralButton("Cerrar", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                Toast.makeText(context, "Selección neutral", Toast.LENGTH_SHORT).show();
                //dismiss();
            }
        });


        return alertDialog.create();
    }
}

File: Kotlin/T6/codigos_ret/navegacion/app/src/main/java/com/example/navegacion/ui/dialog/DialogoConfirmacion.kt
package com.example.navegacion.ui.dialog

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import androidx.navigation.fragment.findNavController
import com.example.navegacion.R
import com.google.firebase.auth.FirebaseAuth

class DialogoConfirmacion: DialogFragment() {

    private lateinit var nombre: String;
    private lateinit var auth: FirebaseAuth

    override fun onAttach(context: Context) {
        super.onAttach(context)
        auth = FirebaseAuth.getInstance()
        nombre = requireArguments().getString("nombre")!!
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder: AlertDialog.Builder = AlertDialog.Builder(requireActivity())
        builder.setTitle("Registro completo")
        builder.setMessage("Enhobuena ${nombre}, ¿quieres logearte directamente?")
        builder.setPositiveButton("SI"){_,_->
            // login al fragment main
            findNavController().navigate(R.id.action_dialogoConfirmacion_to_fragmentMain)
        }
        builder.setNegativeButton("NO"){_,_->
            // deslogeo
            auth.signOut()
            findNavController().navigate(R.id.action_dialogoConfirmacion_to_fragmentLogin)
        }
        return builder.create()
    }
}